src/app/cli/src/services/config/models.rs

```diff
@@ -836,6 +843,216 @@ impl IdentityConfig {
+            })),
+        }
+    }
+}
+
+impl Default for SearchConfig {
+    fn default() -> Self {
+        Self {
+            indexer: Some(SearchIndexerConfig::default()),
+            embeddings_chunker: Some(EmbeddingsChunkerConfig::Simple(
+                EmbeddingsChunkerConfigSimple::default(),
+            )),
+            embeddings_encoder: Some(EmbeddingsEncoderConfig::OpenAi(
+                EmbeddingsEncoderConfigOpenAi::default(),
+            )),
+            vector_repo: Some(VectorRepoConfig::QdrantContainer(
+                VectorRepoConfigQdrantContainer::default(),
+            )),
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[skip_serializing_none]
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct SearchIndexerConfig {
+    // Whether to clear and re-index on start or use existing vectors if any
+    pub clear_on_start: bool,
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[skip_serializing_none]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "kind")]
+pub enum EmbeddingsChunkerConfig {
+    Simple(EmbeddingsChunkerConfigSimple),
+}
+
+impl Default for EmbeddingsChunkerConfig {
+    fn default() -> Self {
+        Self::Simple(EmbeddingsChunkerConfigSimple::default())
+    }
+}
+
+#[skip_serializing_none]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct EmbeddingsChunkerConfigSimple {
+    // Whether to chunk separately major dataset sections like name, schema, readme, or to combine
+    // them all into one chunk
+    pub split_sections: Option<bool>,
+
+    // Whether to split section content by paragraph
+    pub split_paragraphs: Option<bool>,
+}
+
+impl Default for EmbeddingsChunkerConfigSimple {
+    fn default() -> Self {
+        Self {
+            split_sections: Some(false),
+            split_paragraphs: Some(false),
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[skip_serializing_none]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "kind")]
+pub enum EmbeddingsEncoderConfig {
+    OpenAi(EmbeddingsEncoderConfigOpenAi),
+}
+
+impl Default for EmbeddingsEncoderConfig {
+    fn default() -> Self {
+        Self::OpenAi(EmbeddingsEncoderConfigOpenAi::default())
+    }
+}
+
+#[skip_serializing_none]
+#[derive(Debug, Clone, Merge, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct EmbeddingsEncoderConfigOpenAi {
+    pub url: Option<String>,
```

Comment:
`Option<Url>`

Reply:
I was thinking about this, but then saw Qdrant for example supporting "special URLs" like `:memory:` for in-memory impelmentation. Such value would not parse into `Url` and will be rejected, so in case of external components I thought it's better to leave URL validation to them.